src/edit_json.py

```diff
@@ -0,0 +1,198 @@
+                        do_remove = clean_input(
+                            "Are you sure "
+                            "(completely remove this entry)? (y/n): ")
+
+                        if do_remove.lower() in ['y', 'yes']:
+                            obj.pop(key)
+                            print("Removed " + key + " from script.\n")
+                        else:
+                            print("Aborted.")
+                            sleep(1)
+
+                    elif selection == '4' or selection == "":
+                        pass
+                    else:
+                        raise RuntimeError("Invalid input!")
+
+                else:
+                    print("Select one of the following for the key '" + key + "': \n")
+                    print("1. Modify value")
+                    print("2. Remove from script")
+                    print("3. Continue (no changes)\n")
+
+                    selection = clean_input("\nYour choice: ")
+
+                    if selection == '1':
+                        new_val = clean_input('Enter new value: ')
+                        obj[key] = new_val
+
+                    elif selection == '2':
+                        do_remove = clean_input(
+                            "Are you sure (completely remove this entry)? (y/n): ")
+
+                        if do_remove.lower() in ['y', 'yes']:
+                            obj.pop(key)
+                            print("Removed " + key + " from script.\n")
+                        else:
+                            print("Aborted.")
+                            sleep(1)
+                    elif selection == '3' or selection == "":
+                        pass
+                    else:
+                        raise RuntimeError("Invalid input!")
+                break
+            except RuntimeError:
+                continue
+
+
+def edit_json(script):
+    """
+    Edit existing datapackage.JSON script.
+
+    Usage: retriever edit_json <script_name>
+    Note: Name of script is the dataset name.
+    """
+    json_file = script.replace('-', '_') + '.json'
+    try:
+        contents = json.load(
+            open(os.path.join('scripts', json_file), 'r'))
+    except (IOError, OSError):
+        print("Script not found.")
+        return
+
+    edit_dict(contents, 1)
+
+    file_name = contents['name'] + ".json"
+    file_name = file_name.replace('-', '_')
+    with open(os.path.join('scripts', file_name), 'w') as output_file:
+        json.dump(contents,
+                  output_file,
+                  sort_keys=True,
+                  indent=4,
+                  separators=(',', ': '))
+        output_file.write('\n')
+        print("\nScript written to " + os.path.join('scripts', file_name))
+        output_file.close()
+
+
+if __name__ == '__main__':
+    if len(sys.argv) == 1:
+        raise Exception("no dataset specified.")
+    script = sys.argv[1]
+    edit_json(script)
```

Comment:
The entire file has been shipped from retriever's [`datapackage.py`](https://github.com/weecology/retriever/blob/master/retriever/lib/datapackage.py).